### ISSUE TITLE
Could you add a cache option

### ISSUE THREAD

**Issue by ptempier** [theotherp/nzbhydra#362]
Hello

Considering 
- That indexers have an api limit
- That a user can have multiple nzb related app installed
- That it maybe good to reduce load on free indexers

Would it be possible to add a cache option to the results, possibly the nzb?

So that : 
- If the number of api call limit is reached, we can still provide results from the cache
- We can point all nzb related softs to nzbhydra and let it proxy/cache the results
- There s no server hammering in case the app do a search on startup and we restart it muletiple time to configure it
- There s a single request done if i search once for "my fav movie" looking for movie, then after, for "my fav movie" looking for the soundtrack.

Maybe implement it in the indexer tab:
As an additional local indexer server, with some parameter like  keep data in cache for 6h and size 64Mb.
and / or
Per indexer, like in case no result from this indexer, serve the data from cache the last time it worked. With only some size limit to the cache.

**Comment by theotherp** [comment]
In the early days Hydra had a cache option. Nobody seemed to use it so I removed it. But it was search based, not indexer based. I never thought about the case that an indexer with API limit reached wouldn't return any results and how this affects the calling software. On the other hand, how much overlap to you have between tools and their searches? Let's say you use LL, CP and Sonarr. One searches for books, one for movies and one for shows. You will always want the newest results. If the API limit is reached the searching tool doesn't care about the cached results, it already knows them anyway. From its POV there aren't any new results either way. And CP does other searches for Sonarr.

If I understand you correctly you would want to use cached search results when you do more specific searches (like "my movie" and then "my movie soundtrack" and the second search uses in the first one with an additional keyword). That won't work.

But I think there's an argument to be made for a simple indexer based timeout cache. For free indexers with, say, 20 hits per day, you could define the timeout as 4 hours so that three tools combined will amount to 18 searches per day and not use all the available 20 hits in the first three hours.

**Comment by atyoung** [comment]
@theotherp I think this would make more sense for people who are sharing their hydra setup with other people. Arguably (in terms of proper) this would make nzbhydra ideal for people to share paid or closed accounts, as you could cache say, proxied nzb file grabs (in theory) there by reducing API hits for the same download.

**Comment by theotherp** [comment]
There is disagreement if Hydra should do anything to actively support sharing of indexer accounts. I haven't decided one way or the other.

Either way I'm not sure if I would support proxying NZBs. That's a minor use case and would invite all kinds of trouble. I'll keep it in mind.

**Comment by atyoung** [comment]
@theotherp I agree, just pointing out the use case I've heard most frequently. 

That said, sharing is kind of expected given the limited multiuser functionality that already exists, unless each user sets up their own indexer configuration, which is not the case currently.

**Comment by ptempier** [comment]
Before making the request, I was thinking about requesting multiple api keys and put one per app. Because i also thought some clients were not behaving as configured (making search more often than configured, now i am thinking that this might be due to a search on startup setting and the service keeps restarting) and wanted to track this via which request is made via which api key and thus which app. 

Then thoughty this could be misinterpreted. Hence why i made this request for a cache option which in my opinion would also fix the multiple similar queries issue. In my opinion, a cache would be quite healthy in the ecosystem.

To take advantage of a shared cache with a friend you d need to rent a server or keep your pc on 24/7 and also like the same kind of movies/music/series, and also have very similar download setting and make the search almost at the same time, and also have some basic it skills to setup the whole thing. For it to be interesting you'd need to share it with a lot of peoples and eventually that will cost some money, and eventually you'll make a paying service

Peoples wanting to provide a payed service based on a cache would need to develop a lot of things,  so hydra might not be useful for their use case. And who would pay for old results?

i only use free stuff, so i use multiple indexers with a typically very low api hit. Then set up only hydra as indexer in the various apps (avoid to change every app when one indexer disappears, which happens quite often.).
